### PR TITLE
Add kani test and remove TODO

### PR DESCRIPTION
**Pull request by tcharding**
Add a kani test to check `div_rem`.

### CODE REVIEW

**Comment by tcharding** [comment]
I"m a bit rusty on `kani` but this seems pretty straight forward. CI won't test it till we merge but it appears to run ok locally.

File: bitcoin/src/pow.rs [hunk 1]
```diff
@@ -1744,4 +1743,13 @@ mod verification {
 
         let _ = x.mul_u64(y);
     }
+
+    #[kani::unwind(5)]   // I can't remember exactly why we need this.
```
Comment (Kixunil) [inline bitcoin/src/pow.rs:1747]: I believe the code uses a loop of length 5 somewhere (or more likely 4).

**Review by apoelstra — APPROVED** [review]
ACK 66352cba98c81024c7fde264f672b1cab01c47e1

**Comment by apoelstra** [comment]
This fails right now because we have a `debug_assert` in our `Mul` implementation that checks for overflow.

Do we want to modify the test or what?

**Comment by tcharding** [comment]
I just clobbered it, onwards and upwards.

**Comment by tcharding** [comment]
That does mean we have lost the `TODO` but since there are loads of things we can do to improve testing I don't think loosing one todo comment is a big loss.

**Pull request merged**